compiler/rustc_middle/src/middle/stability.rs

```diff
@@ -126,36 +128,14 @@ pub fn report_unstable(
 /// Checks whether an item marked with `deprecated(since="X")` is currently
 /// deprecated (i.e., whether X is not greater than the current rustc version).
 pub fn deprecation_in_effect(depr: &Deprecation) -> bool {
-    let is_since_rustc_version = depr.is_since_rustc_version;
-    let since = depr.since.as_ref().map(Symbol::as_str);
-
-    if !is_since_rustc_version {
+    match depr.since {
+        Some(DeprecatedSince::RustcVersion(since)) => since <= RustcVersion::CURRENT,
+        Some(DeprecatedSince::Future) => false,
         // The `since` field doesn't have semantic purpose without `#![staged_api]`.
-        return true;
+        Some(DeprecatedSince::Symbol(_)) => true,
+        // Assume deprecation is in effect if "since" field is missing.
+        None => true,
     }
-
-    if let Some(since) = since {
-        if since == "TBD" {
-            return false;
-        }
-
-        // We ignore non-integer components of the version (e.g., "nightly").
-        let since: Vec<u16> =
-            since.split(|c| c == '.' || c == '-').flat_map(|s| s.parse()).collect();
-
-        // We simply treat invalid `since` attributes as relating to a previous
-        // Rust version, thus always displaying the warning.
-        if since.len() != 3 {
-            return true;
-        }
-
-        let rustc = RustcVersion::CURRENT;
-        return since.as_slice() <= &[rustc.major, rustc.minor, rustc.patch];
-    };
-
-    // Assume deprecation is in effect if "since" field is missing
-    // or if we can't determine the current Rust version.
```

Comment:
The "or if we can't determine the current Rust version" part of this comment should have been removed by #117256.